packages/entity/src/EntityQueryContext.ts

```diff
@@ -132,8 +149,13 @@ export class EntityTransactionalQueryContext extends EntityQueryContext {
   }
 
   async runInTransactionIfNotInTransactionAsync<T>(
-    transactionScope: (queryContext: EntityTransactionalQueryContext) => Promise<T>
+    transactionScope: (queryContext: EntityTransactionalQueryContext) => Promise<T>,
+    transactionConfig?: TransactionConfig
   ): Promise<T> {
+    assert(
+      transactionConfig === undefined,
+      'should not pass transactionConfig to an already created transaction'
```

Comment:
Is this slightly more accurate?
```suggestion
      'Should not pass transactionConfig to a nested transaction'
```

Reply:
I think an isolation level can be set as long as no queries have been run in the transaction yet, but this is probably the safest way of doing this without adding extra checks.

Reply:
That's accurate but the failure mode is dangerous. Specifically, Postgres just prints a warning. So you could have code that intends to set the isolation level to something strict (like wanting a serializability guarantee) and it could end up a no-op and you'd never know.

Reply:
Yeah, my thought was to either keep track of a given `transactionConfig` in a transaction and allow it to be set as long as no other queries had been run, *or* check for any pgsql warnings/output. Both seem overkill here.